File: admin/lib/categories.ts
// Canonical item categories for the admin panel — single source of truth.
// Keep in sync with backend/app/core/categories.py. Adding a category later =
// one entry here.

export const CATEGORIES = [
  "men",
  "women",
  "kids",
  "unisex",
  "home",
  "footwear",
  "accessories",
  "other",
] as const;

export type Category = (typeof CATEGORIES)[number];

export const CATEGORY_STYLES: Record<string, string> = {
  men: "bg-blue-100 text-blue-700",
  women: "bg-pink-100 text-pink-700",
  kids: "bg-purple-100 text-purple-700",
  home: "bg-teal-100 text-teal-700",
  footwear: "bg-amber-100 text-amber-700",
  accessories: "bg-rose-100 text-rose-700",
  unisex: "bg-gray-100 text-gray-500",
  other: "bg-gray-100 text-gray-500",
};

export function categoryStyle(key: string): string {
  return CATEGORY_STYLES[key] || CATEGORY_STYLES.unisex;
}

File: backend/app/core/categories.py
"""Canonical item-category list — the single source of truth for the backend.

Adding a new category later is a one-line change here (and one line in each
app's category constant). Apps keep their own presentation metadata (icons,
colours, home-tile flags); the backend only needs the keys for validation.
"""

# Order matters only for display fallbacks; membership is what's enforced.
ITEM_CATEGORIES = [
    "men",
    "women",
    "kids",
    "unisex",       # default — generic apparel worn by everyone
    "home",         # household linens & furnishings (curtains, bedsheets, sofa…)
    "footwear",     # shoes
    "accessories",  # bags, soft toys, caps, ties…
    "other",        # catch-all
]

DEFAULT_CATEGORY = "unisex"


def is_valid(category: str) -> bool:
    return category in ITEM_CATEGORIES


def normalize(category) -> str:
    """Return a valid category, falling back to the default."""
    return category if category in ITEM_CATEGORIES else DEFAULT_CATEGORY
